LoopKitUI/Views/ExpandableDatePicker.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  ExpandableDatePicker.swift
+//  LoopKitUI
+//
+//  Created by Anna Quinlan on 8/12/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import SwiftUI
+
+public struct ExpandableDatePicker: View {
+    @State var dateShouldExpand = false
+    @Binding var date: Date
+    let placeholderText: String
+    let pickerRange: ClosedRange<Date>
+    @State public var userDidTap: Bool = false
+    
+    public init (
+        with date: Binding<Date>,
+        pickerRange: ClosedRange<Date>? = nil,
+        placeholderText: String = ""
+    ) {
+        _date = date
+        self.placeholderText = placeholderText
+        
+        let today = Date()
+        self.pickerRange = pickerRange ?? today.addingTimeInterval(-.hours(24))...today
```

Comment:
Since `.hours` is internal, we can't have it be a default argument in the initializer